engine/kv_engine.cpp

```diff
@@ -1720,3 +1736,193 @@ Status KVEngine::RestoreDlistRecords(DLRecord *pmp_record) {
 }
 
 } // namespace KVDK_NAMESPACE
+
+namespace KVDK_NAMESPACE {
+std::unique_ptr<Queue> KVEngine::createQueue(StringView const collection_name) {
+  std::uint64_t ts = get_timestamp();
+  uint64_t id = list_id_.fetch_add(1);
+  std::string name(collection_name.data(), collection_name.size());
+  std::unique_ptr<Queue> queue_uptr =
+      std::make_unique<Queue>(pmem_allocator_.get(), name, id, ts);
+  return queue_uptr;
```

Comment:
std::move(queue_uptr)

Reply:
You don't need to move a rvalue returned by std::make_unique. The compiler knows the return value from std::make_unique is a rvalue and calls std::unique_ptr(std::unique_ptr&&)(The move constructor is called, not the move assignment operator).
I should use curly-brace initializer to make things clear.

Reply:
see: https://stackoverflow.com/questions/33971318/the-correct-way-of-returning-stdunique-ptr-to-an-object-of-polymorphic-class. You can't assume that compiler always do this optimization, different compiler version maybe have different result

Reply:
Well, that looks funny

Reply:
done